### ISSUE TITLE
Document using `zini-pthash` from `build.zig`

### ISSUE THREAD

**Issue by malcolmstill** [judofyr/zini#4]
I was playing around yesterday with getting a `*.pth` file generated as part of a `build.zig` rather than invoking `zini-pthash` manually. Now, that was using a custom binary for my particular application, however, if `zini-pthash` suffices then it is straightforward to use `zini-pthash` directly without a separate implementation.

Let's say you are building a program that will consume (i.e. call `.get()`) a `*.pth` file, so it depends on `zini` (now via `build.zig.zon` as per https://github.com/judofyr/zini/pull/2). Your `build.zig` will look something like:

```zig
    const zini = b.dependency("zini", .{ .target = target, .optimize = optimize });

    const exe = b.addExecutable(.{
        .name = "myPthConsumer",
        .root_source_file = .{ .path = "src/main.zig" },
        .target = target,
        .optimize = optimize,
    });

    exe.root_module.addImport("zini", zini.module("zini"));
```

Now to run `zini-pthash` to generate and make available the `.pth` at build time you can do:

```zig
    const zini = b.dependency("zini", .{ .target = target, .optimize = optimize });

    // Build our .pth file
    const zini_build_exe = zini.artifact("zini-pthash"); // Get a reference to the `b.addExecutable` for `zini-pthash`
    const builder_step = b.addRunArtifact(zini_build_exe); // Run `zini-pthash` during build
    builder_step.addArg("build");
    builder_step.addArg("-i");
    builder_step.addFileArg(.{ .path = "input.txt" });
    builder_step.addArg("-o");
    // zig will generate a path somewhere in `zig-cache` for this output:
    const pth_output = builder_step.addOutputFileArg("out.pth"); 

    const exe = b.addExecutable(.{
        .name = "myPthConsumer",
        .root_source_file = .{ .path = "src/main.zig" },
        .target = target,
        .optimize = optimize,
    });

    exe.root_module.addImport("zini", zini.module("zini"));
    // Make the building of `myPthConsumer` depend on `builder_step` and make `out.pth`
    // available to `myPthConsumer` via `@embedFile("pth")`:
    exe.root_module.addAnonymousImport("pth", .{ .root_source_file = pth_output }); 
```

It might be worth documenting this alongside the rest of the information under `Usage` in `README.md`, as I think this would be typical of any use of `zini` as a library?

**Comment by judofyr** [comment]
Interesting! Originally when I wrote Zini I expected that people would only directly use it as a library. `zini-pthash` was there mainly as a way of providing an example (documentation) and to validate that it was working end-to-end. The file reading/writing was never really supposed to be a proper file format (there's no file magic or validation or anything). Not quite sure how I feel about it. It suddenly becomes more of a "public API". Not that it's a big problem. It's not like I intend to change these things a lot.

This concept of embedding it directly into the binary is very interesting! Maybe we could make that even smoother in some way?

**Comment by malcolmstill** [comment]
> Not quite sure how I feel about it. It suddenly becomes more of a "public API"

Yeah, that's fair. It's not difficult for someone to implement whatever they need in terms of building the `.pth` (perhaps with `zini-pthash` as a reference) and it's also quite likely that they need to do some bespoke anyway.

> This concept of embedding it directly into the binary is very interesting!

Ah yeah, I'm being slightly blinded by what I'm doing where I'm happy to build in the data (at the cost of requiring a rebuild of the consuming program when the data changes)...obviously, users may prefer loading the `.pth` files via the file system. I think there are a couple of these "recipes" that might be worth highlighting.
